src/audio/module_adapter/module_adapter_ipc3.c

```diff
@@ -79,6 +79,9 @@ int module_adapter_init_data(struct comp_dev *dev,
 			return ret;
 		}
 		dst->init_data = dst->data;
+	} else if (size == 0) {
+		/* valid case for mixer */
+		return 0;
```

Comment:
if this is the case, do we still need keep line 85 - 87?

Reply:
@btian1 true, size can't be negative. So maybe we just do a full revert.

It does open questions how to address the original commit e847c8b270150353551d4720e382502c9a916496 . Do we need a SRC specific patch?

Reply:
I agree @kv2019i , I'll do a full revert of this https://github.com/thesofproject/sof/commit/e847c8b270150353551d4720e382502c9a916496, but the revert doesn't apply cleanly now because of the ipc3/ipc4 split.

I've kept the check for negative size to not break something on Intel targets - even though the explanation for adding that commit is limited.